x-pack/plugins/security_solution/public/common/components/user_privileges/index.tsx

```diff
@@ -6,32 +6,39 @@
  */
 
 import React, { createContext, useContext } from 'react';
-import { useFetchDetectionEnginePrivileges } from './use_fetch_detection_engine_privileges';
-import { useFetchListPrivileges } from './use_fetch_list_privileges';
+import { useFetchDetectionEnginePrivileges } from '../../../detections/components/user_privileges/use_fetch_detection_engine_privileges';
+import { useFetchListPrivileges } from '../../../detections/components/user_privileges/use_fetch_list_privileges';
+import { EndpointPrivileges, useEndpointPrivileges } from './use_endpoint_privileges';
 
 export interface UserPrivilegesState {
   listPrivileges: ReturnType<typeof useFetchListPrivileges>;
   detectionEnginePrivileges: ReturnType<typeof useFetchDetectionEnginePrivileges>;
+  endpointPrivileges: EndpointPrivileges;
 }
 
-const UserPrivilegesContext = createContext<UserPrivilegesState>({
+export const initialUserPrivilegesState = (): UserPrivilegesState => ({
```

Comment:
❓  should the return value of the hook `useUserPrivileges` (below line49) return a `ReadOnly<>` response?  The value returns (I think) should not be mutable - i know that should def. be the case for the endpoint privileges object, but am not sure about the other two.

Reply:
Yes, `listPrivileges` and  `detectionEnginePrivileges` should also be readonly.